lib/SILOptimizer/Mandatory/DIMemoryUseCollectorOwnership.cpp

```diff
@@ -950,6 +950,15 @@ void ElementUseCollector::collectUses(SILValue Pointer, unsigned BaseEltNo) {
       continue;
     }
 
+    // unchecked_take_enum_data_addr takes the address of the payload of the
+    // optional, which could be used to update the payload. So, visit the
+    // users of this instruction and ensure that there are no overwrites to an
+    // immutable optional. Note that this special handling is for checking
+    // immutability and is not for checking initialization before use.
+    if (auto *enumDataAddr = dyn_cast<UncheckedTakeEnumDataAddrInst>(User)) {
+      collectUses(enumDataAddr, BaseEltNo);
+    }
```

Comment:
@gottesmm  Looks like it is possible add a "continue" after the call to 'collectUses'. This will make the code similar to other cases in the function. However, this implies that we would *not* be adding "take_enum_data_addr" instruction to the users list but would instead be collecting its users. WDYT? With this modification is there any benefit in retaining 'take_enum_data_addr' instruction?